contrib/scrooge/src/python/pants/contrib/scrooge/tasks/scrooge_gen.py

```diff
@@ -161,11 +172,13 @@ def gen(self, partial_cmd, target, target_workdir):
     for lhs, rhs in partial_cmd.namespace_map:
       args.extend(['--namespace-map', '%s=%s' % (lhs, rhs)])
 
-    if partial_cmd.rpc_style == 'ostrich':
-      args.append('--finagle')
-      args.append('--ostrich')
-    elif partial_cmd.rpc_style == 'finagle':
-      args.append('--finagle')
+    # ignore rpc_style if we think compiler_args is setting it
```

Comment:
As mentioned above, I think this arg creation should move before `PartialCmd` creation.